File: powermul.py
import math,sys
from sys import stdin
from math import pow 
t = sys.stdin.readline()
t = int (t)
while(t):
	n,m,q = map(int, sys.stdin.readline().split(' '))
	while(q):
		sum1 = 1
		i = sys.stdin.readline()
		i = int(i)
		if(n-i>i):
			sum11,sum12,sum13=1,1,1
			for j in range(1,n+1):
				sum11 = sum11*pow(j,j)
				if(j<=i):
					sum12 = sum12*pow(j,j)
				if(j<=n-i):
					sum13 = sum13*pow(j,j)
		sum1 =  (sum11/(sum12*sum13))%m
		print int(sum1)
		q=q-1
	t = t-1

File: powermul.c
#include<stdio.h>
#include<math.h>
int main()
{
	long long int t,sum1,sum2,sum3,n,m,p,q,sum,j;
	sum1 = 1;
	sum2 = 1;
	sum3 = 1;
	scanf("%lld",&t);
	while(t--)
	{
		scanf("%lld %lld %lld",&n,&m,&q);
		for(j=1;j<n+1;j++)
			sum1 = sum1*pow(j,j);
		while(q--)
		{
			sum2 = 1;
			sum3 = 1;
			scanf("%lld",&p);
			for(j=1;j<p+1;j++)
			{
				sum2 = sum2*pow(j,j);
			// for(j=1;j<n-p+1;j++)
				sum3 = sum3*pow(j,j);
			}
			sum = (sum1/(sum2*sum3))%m;
			printf("%lld\n",sum);
		}
	}
	return 0;
}